Create user model and run Alembic migration

MeghaMuralidharan opened the pull request:
Define Router: Create the FastAPI router file for user endpoints.
Create User: Implement the POST endpoint to register a new user (with password hashing).
Read User: Implement the GET endpoint to fetch a user by ID/username.
Wire Up: Include the new router in your main application (app/main.py).

MeghaMuralidharan committed:
Initial commit

the-chosen-reject committed:
Merge pull request #49 from Nexeor-dev-space/feat/1-MeghaMuralidharan

Initialize FastAPI project with Poetry dependencies

MeghaMuralidharan committed:
Issue 2: Organized FastAPI project folder structure

the-chosen-reject committed:
Merge pull request #51 from Nexeor-dev-space/feat/2-MeghaMuralidharan

Organize FastAPI project folder layout

MeghaMuralidharan committed:
Issue 3: Add database configuration with SQLAlchemy and pydantic-settings

MeghaMuralidharan committed:
Refactored Settings to use Pydantic v2 (SettingsConfigDict) and moved to config.py

MeghaMuralidharan committed:
feat: Completed Issue 4 - User Model, Schemas, Hashing, and Initial Migration

iinava commented on alembic.ini:
```diff
@@ -0,0 +1,147 @@
+# database URL.  This is consumed by the user-maintained env.py script only.
+# other means of configuring database URLs may be customized within the env.py
+# file.
+sqlalchemy.url = postgresql://admin:123456@localhost:5432/admin_db
```
remove this hardcoded db url

iinava commented on app/core/config.py:
```diff
@@ -0,0 +1,18 @@
+    JWT_SECRET_KEY: str
+    ACCESS_TOKEN_EXPIRE_MINUTES: int
+    REFRESH_TOKEN_EXPIRE_MINUTES: int
+    JWT_ALGORITHM: str
```
same things  are duplicated in settings.py

iinava commented on app/schemas/user.py:
```diff
@@ -0,0 +1,19 @@
+    fullname: str = Field(min_length=3, max_length=100)
+
+# Schema used when creating a new user (requires password)
+class UserCreate(UserBase):
```
user.py and users.py similiar duplicated schema files

iinava commented on app/db/session.py:
```diff
@@ -0,0 +1,8 @@
+from app.db.database import async_session
```
database.py exports AsyncSessionLocal, but session.py tries to import async_session (which doesn't exist). This will crash at runtime!

MeghaMuralidharan committed:
Update alembic.ini

removed this hardcoded db url

iinava commented on app/core/config.py:
```diff
@@ -0,0 +1,18 @@
+    ASYNC_DATABASE_URL: str
+    JWT_SECRET_KEY: str
+    ACCESS_TOKEN_EXPIRE_MINUTES: int
+    REFRESH_TOKEN_EXPIRE_MINUTES: int
```
duplicate in settings.py

MeghaMuralidharan committed:
feat: Complete issue 4 - User model and Alembic migration

- Created User model with id, username, fullname, hashed_password
- Generated and applied Alembic migration (fb87f0c5bde5_add_user_model.py)
- Added password hashing using passlib[bcrypt]
- Created UserCreate and UserRead schemas
- Fixed database session configuration
- Updated alembic configuration for SQLite
- Added .env file with database configuration
- Renamed models/users.py to models/user.py for consistency

the-chosen-reject commented:
Closing PR.
